deepchem/models/torch_models/normalizing_flows_pytorch.py

```diff
@@ -1,2 +1,89 @@
 """Normalizing flows for transforming probability distributions using PyTorch.
 """
+import torch
+from torch import nn
+from typing import Sequence, Tuple
+
+
+class NormalizingFlow(nn.Module):
+  """Normalizing flows are widley used to perform generative models.
+  This algorithm gives advantages over variational autoencoders (VAE) because
+  of ease in sampling by applying invertible transformations
+  (Frey, Gadepally, & Ramsundar, 2022).
+
+  Example
+  --------
+  >>> import deepchem as dc
+  >>> from deepchem.models.torch_models.layers import Affine
+  >>> from deepchem.models.torch_models.normalizing_flows_pytorch import NormalizingFlow
+  >>> import torch
+  >>> from torch.distributions import MultivariateNormal
+  >>> # initialize the transformation layer's parameters
+  >>> dim = 2
+  >>> samples = 96
+  >>> transforms = [Affine(dim)]
+  >>> distribution = MultivariateNormal(torch.zeros(dim), torch.eye(dim))
+  >>> # initialize normalizing flow model
+  >>> model = NormalizingFlow(transforms, distribution, dim)
+  >>> # evaluate the log_prob when applying the transformation layers
+  >>> tensor = distribution.sample(torch.Size((samples, dim)))
+  >>> len(model.log_prob(tensor))
+  96
+  >>> # evaluates the the sampling method and its log_prob
+  >>> len(model.sample(samples))
+  2
+
+  """
+
+  def __init__(self, transform: Sequence, base_distribution, dim: int) -> None:
```

Comment:
Yes, so I have some time on deciding which type hint class to write here. I looked that some CI fails raised If used `torch.Tensor`. This is mainly because `base_distribution` is a `torch.distributions` object. I will write `torch.distributions`, does this make sense 🤔?

Reply:
Awesome!! I will Include this type hint as well as the `Transforms` class and FastFlow reference on my next PR 🙌🏼